lib/timeline/frame_flame_chart.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../ui/elements.dart';
+import '../ui/primer.dart';
+import 'timeline_protocol.dart';
+
+class FrameFlameChart extends CoreElement {
+  FrameFlameChart() : super('div') {
+    layoutVertical();
+    flex();
+
+    // TODO(devoncarew): listen to tab changes
+    content = div(c: 'frame-timeline')..flex();
+
+    final PTabNav tabNav = PTabNav(<PTabNavTab>[
+      PTabNavTab('Frame timeline'),
+      PTabNavTab('Widget build info'),
+      PTabNavTab('Skia picture'),
+    ]);
+
+    add(<CoreElement>[
+      tabNav,
+      content,
+    ]);
+
+    content.element.style.whiteSpace = 'pre';
+    content.element.style.overflow = 'scroll';
+  }
+
+  TimelineFrameData data;
+  CoreElement content;
+
+  void updateData(TimelineFrameData data) {
+    this.data = data;
+
+    content.clear();
+
+//    if (data != null) {
```

Comment:
remove this commented out debugging code or add a private flag to this file so others can turn it on and off as needed
e.g.
bool _debugTraceEvents = false;

Reply:
Done.